sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlanVisitor.scala

```diff
@@ -47,6 +49,15 @@ trait LogicalPlanVisitor[T] {
 
   def default(p: LogicalPlan): T
 
+  def visitSubqueryExpression(p: LogicalPlan): LogicalPlan = {
+    p.transformExpressionsDown {
+      case subqueryExpression: SubqueryExpression =>
+        subqueryExpression.plan.stats
```

Comment:
Can you also add a comment that this is to trigger stats propagation?

Reply:
> Can you also add a comment that this is to trigger stats propagation?

Done